polaris.shopify.com/src/components/IconsPage/IconsPage.tsx

```diff
@@ -1,310 +1,145 @@
+    router.push({ query });
+  };
 
-    return () => {
-      router.events.off("routeChangeComplete", setSelectedIconBasedOnParam);
-    };
-  }, [setSelectedIconBasedOnParam, router.events]);
+  const handleRemoveIcon = () => {
+    const query: { q?: string } = {};
+    if (searchText) query.q = searchText;
+    router.push({ query });
+  };
 
   return (
     <Container className={styles.IconsPage}>
-      <PageMeta title="Icons" />
-
-      <div className={styles.Filter}>
-        <h1>Icons</h1>
-      </div>
-
-      <div className={styles.SplitLayout}>
+      <PageMeta title={pageTitle} />
+      <h1>Icons</h1>
+      <div className={!useModal ? styles.SplitLayout : ""}>
         <div className={styles.IconGrids}>
-          <div className={styles.TextField}>
-            <TextField
-              value={filterString}
-              onChange={(value) => setFilterString(value)}
-              placeholder="Filter icons"
+          <SearchField
+            value={searchText}
+            onChange={(value) => handleSearchChange(value)}
+            placeholder="Search icons"
+          />
+          {Object.keys(matchingMajor).length !== 0 && (
+            <IconGrid
+              title="Major icons"
+              icons={matchingMajor}
+              activeIcon={activeIcon}
+              query={searchText}
             />
-          </div>
-
-          {majorIcons.length > 0 && (
-            <>
-              <div className={styles.SectionHeading}>
-                <p>
-                  <b>Major icons</b>
-                </p>
-              </div>
-              <IconGrid>
-                {majorIcons.map((icon) => (
-                  <IconGrid.Item
-                    key={icon.name}
-                    icon={icon}
-                    onClick={() => {
-                      window.history.pushState(
-                        {},
-                        "",
-                        `?icon=${icon.fileName}`
-                      );
-                      getSelectedIcon(icon);
-                    }}
-                    isSelected={selectedIcon.fileName === icon.fileName}
-                  />
-                ))}
-              </IconGrid>
-            </>
           )}
-
-          {minorIcons.length > 0 && (
-            <>
-              <div className={styles.SectionHeading}>
+          {Object.keys(matchingMinor).length !== 0 && (
+            <IconGrid
+              title="Minor icons"
+              icons={matchingMinor}
+              activeIcon={activeIcon}
+              query={searchText}
+            />
+          )}
+          {Object.keys(matchingMajor).length === 0 &&
+          Object.keys(matchingMinor).length === 0 ? (
+            <div className={styles.NoSearchResults}>
+              <span style={{ opacity: 0.25 }}>
+                <Image
+                  src="/icons/SearchMajor.svg"
+                  width={20}
+                  height={20}
+                  alt=""
```

Comment:
Question for the site in general: Should icons have alt texts? My read on a11y recommendations is that we should set `alt=""` if they are purely decorative and the context is enough to understand what's going on.

Reply:
`<Image>` from next/js requires it. I think we should set an alt here.